tests/test_backend.py

```diff
@@ -391,108 +329,95 @@ def test_file(tmpdir, local_file, remote_file, version, ext, backend):
 )
 @pytest.mark.parametrize(
     'backend',
-    [
-        audbackend.FileSystem(
-            pytest.FILE_SYSTEM_HOST,
-            pytest.REPOSITORY_NAME,
-        ),
-        audbackend.Artifactory(
-            pytest.ARTIFACTORY_HOST,
-            pytest.REPOSITORY_NAME,
-        ),
-    ]
+    pytest.BACKENDS,
+    indirect=True,
 )
 def test_glob(tmpdir, files, pattern, folder, expected, backend):
 
+    src_path = audeer.path(tmpdir, '~')
+    audeer.touch(src_path)
+
     paths = []
-    for file in files:
-        local_file = os.path.join(tmpdir, file)
-        audeer.mkdir(os.path.dirname(local_file))
-        with open(local_file, 'w'):
-            pass
-        remote_file = backend.join(
-            pytest.ID,
-            'test_glob',
-            file,
-        )
+    for dst_path in files:
         paths.append(
             backend.put_file(
-                local_file,
-                remote_file,
+                src_path,
+                dst_path,
                 '1.0.0',
             )
         )
 
     expected = [
         backend._path(
-            backend.join(
-                pytest.ID,
-                'test_glob',
-                *x.split(backend.sep),
-            ),
+            file,
             '1.0.0',
             '.ext',
         )
-        for x in expected
+        for file in expected
     ]
 
     assert sorted(expected) == sorted(backend.glob(pattern, folder=folder))
 
 
-@pytest.mark.parametrize(
-    'path, content, expected_content',
-    [
-        (
-            'folder1',
-            ['file.txt', 'folder/abc.txt'],
-            ['file', 'folder'],
-        ),
-    ],
-)
 @pytest.mark.parametrize(
     'backend',
-    [
-        audbackend.FileSystem(
-            pytest.FILE_SYSTEM_HOST,
-            pytest.REPOSITORY_NAME,
-        ),
-        audbackend.Artifactory(
-            pytest.ARTIFACTORY_HOST,
-            pytest.REPOSITORY_NAME,
-        ),
-    ]
+    pytest.BACKENDS,
+    indirect=True,
 )
-def test_ls(tmpdir, path, content, expected_content, backend):
+def test_ls(tmpdir, backend):
 
-    backend_path = backend.join(
-        pytest.ID,
-        'test_ls',
-        path,
-    )
+    assert backend.ls() == []
+    assert backend.ls('/') == []
```

Comment:
This feels still a little bit strange, but I guess `/`, the root of the backend,  is officially not a folder and it makes sense to return an empty list and not raise a `FileNotFoundError`. I think we should add a section to the docstring, stating that this is the only case you can get an empty list?

Reply:
We now say:

![image](https://user-images.githubusercontent.com/10383417/207637058-25be10e5-91e9-420b-800e-a836bb796fa8.png)